scipy/stats/tests/test_rank.py

```diff
@@ -81,6 +81,11 @@ def test_empty(self):
         assert_array_equal(r, np.array([], dtype=np.float64))
         r = rankdata([])
         assert_array_equal(r, np.array([], dtype=np.float64))
+        b = np.empty((0, 1, 2), dtype=int)
+        r = rankdata(b, axis=None)
+        assert_array_equal(r, np.array([], dtype=np.float64))
+        r = rankdata(b, axis=0)
+        assert_array_equal(r, np.empty_like(b, dtype=np.float64))
```

Comment:
- It looks like the original test was written as though `assert_array_equal` cared about `dtype`s, but it doesn't check by default. Consider passing `strict=True`.
- The new part of this test, at least, looks like a good candidate for parametrization. To prevent future regressions, why not test for `axis=1` and `axis=2`?

Reply:
Done, thanks!